lib/Data/Chronicle/Subscriber.pm

```diff
@@ -0,0 +1,168 @@
+
+=item B<Persistent>
+
+In addition to caching every incoming data, it is also stored in PostgreSQL for future retrieval.
+
+=item B<Transparent>
+
+This modules hides all the details about distribution, caching, database structure and ... from developer. He only needs to call a method
+to save data and another method to retrieve it. All the underlying complexities are handled by the module.
+
+=back
+
+=head1 Example
+
+ my $d = get_some_log_data();
+
+ my $chronicle_w = Data::Chronicle::Writer->new(
+    cache_writer   => $writer,
+    dbic           => $dbic,
+    ttl            => 86400,
+    publish_on_set => 1);
+
+ my $chronicle_s = Data::Chronicle::Subscriber->new(
+    cache_reader => $subscriber);
+
+ #subscribe to changes to syslog
+ $chronicle_s->subscribe("log_files", "syslog", sub { print 'Hello World' });
+
+ #store data into Chronicle - each time we call `set` it will also call
+ #the stored subroutine
+ $chronicle_w->set("log_files", "syslog", $d);
+ 
+ #unsubscribe to changes to syslog
+ $chronicle_s->subscribe("log_files", "syslog"); 
+
+=cut
+
+=head1 Members
+
+=head2 cache_subscriber
+
+cache_subscriber should be an object of RedisDB.
+
+=cut
+
+has 'cache_subscriber' => (
+    is      => 'ro',
+    default => undef,
+);
+
+=head2 subscribe
+
+Example:
+
+    $chronicle_writer->subscribe("category1", "name1", $code_ref);
+
+=cut
+
+sub subscribe {
+    my ($self, $category, $name, $callback) = @_;
+    die 'Subscription requires a coderef' if ref $callback ne 'CODE';
+
+    my $key = $self->_generate_key($category, $name);
+    return $self->cache_subscriber->subscribe($key, $callback);
+}
+
+=head2 unsubscribe
+
+Example:
+
+    $chronicle_writer->unsubscribe("category1", "name1");
+
+=cut
+
+sub unsubscribe {
+    my ($self, $category, $name) = @_;
+
+    my $key = $self->_generate_key($category, $name);
+    return $self->cache_subscriber->unsubscribe($key);
+}
+
+sub _generate_key {
+    my ($self, $category, $name) = @_;
+    return $category . '::' . $name;
+}
+
+no Moose;
+
+=head1 AUTHOR
+
+Binary.com, C<< <support at binary.com> >>
+
+=head1 BUGS
+
+Please report any bugs or feature requests to C<bug-data-chronicle at rt.cpan.org>, or through
+the web interface at L<http://rt.cpan.org/NoAuth/ReportBug.html?Queue=Data-Chronicle>.  I will be notified, and then you'll
+automatically be notified of progress on your bug as I make changes.
+
+
+
+
+=head1 SUPPORT
+
+You can find documentation for this module with the perldoc command.
+
+    perldoc Data::Chronicle::Subscriber
+
+
+You can also look for information at:
+
+=over 4
+
+=item * RT: CPAN's request tracker (report bugs here)
+
+L<http://rt.cpan.org/NoAuth/Bugs.html?Dist=Data-Chronicle>
+
+=item * AnnoCPAN: Annotated CPAN documentation
+
+L<http://annocpan.org/dist/Data-Chronicle>
+
+=item * CPAN Ratings
+
+L<http://cpanratings.perl.org/d/Data-Chronicle>
+
+=item * Search CPAN
+
+L<http://search.cpan.org/dist/Data-Chronicle/>
```

Comment:
This looks like standard boilerplate from one of the automated module generators.

search.cpan.org is now no more, and is redirecting to metacpan.org. This should probably link to

    L<https://metacpan.org/release/Data-Chronicle/>

Ditto in any other files.